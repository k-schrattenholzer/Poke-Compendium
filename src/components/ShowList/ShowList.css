.showList-Container {
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
}
