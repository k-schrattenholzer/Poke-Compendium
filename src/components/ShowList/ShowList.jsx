// import { ShowsWithImg } from "../../utils/utils.js";
import Show from "../Show/Show.jsx";
import "./ShowList.css";

export default function ShowList({ shows }) {

  console.log(shows)

  return (
    <ul aria-label="showList" className="showList-Container">
      {shows.map(( show ) => {          
        return (         
            <li key={show.id}>
              <Show show={ show } />
            </li>
        )
      })}
      
    </ul>
  );
}
