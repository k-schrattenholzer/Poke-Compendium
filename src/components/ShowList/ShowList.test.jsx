import { screen, render } from "@testing-library/react";
import ShowList from './ShowList.jsx';

it('should render showList', () => {
  render (
    <ShowList shows={[
      {
        id: 44,
        name: 'Scorpion',
        lang: 'English',
        premiere: '2014-09-22',
        image: 'url',
        summary: 'inspired by a true story, is a high-octane drama about eccentric genius Walter OBrien and his team of brilliant misfits who comprise the last line of defense against complex, high-tech threats of the modern age. As Homeland Securitys new think tank, OBriens team includes Toby Curtis an expert behaviorist who can read anyone; Happy Quinn, a mechanical prodigy; and Sylvester Dodd, a statistics guru. Pooling their extensive technological knowledge to solve mind-boggling predicaments amazes federal agent Cabe Gallo, who shares a harrowing history with OBrien. However, while this socially awkward group is comfortable with each others humor and quirks, life outside their circle confounds them, so they rely on Paige Dineen, who has a young, gifted son, to translate the world for them. At last, these nerdy masterminds have found the perfect job',
        url: 'https://www.tvmaze.com/shows/45/ncis-new-orleans',
        genres: [
          'genre1',
          'genre2',
          'genre3'
        ],
      },
      {
        id: 46,
        name: 'Forever',
        lang: 'English',
        premiere: '2014-09-22',
        image: 'https://static.tvmaze.com/uploads/images/medium_portrait/128/322484.jpg',
        summary: 'Doctor Henry Morgan, New York Citys star medical examiner, has a secret. He doesnt just study the dead to solve criminal cases',
        url: 'https://www.tvmaze.com/shows/47/witches-of-east-end',
        genres: [
          "Drama",
          "Fantasy",
          "Romance"
      ],
      }
    ]}/>
  )

  const showList = screen.getByText(/Scorpion/);
  expect(showList).toMatchSnapshot()
})