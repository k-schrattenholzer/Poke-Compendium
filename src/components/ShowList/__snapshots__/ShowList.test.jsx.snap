// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render showList 1`] = `
<h4>
  Scorpion
</h4>
`;
